### PR TITLE
Add Sound Null Safety

### PR DESCRIPTION
**Pull request by DebkanchanSamadder**
1. Added sound null safety
2. Updated Dart SDK
3. Updated .gitignore to ignore `flutter_export_enviorment.sh`
4. Removed unused I18n generated file
5. Version bump
6. Update the whole iOS system to new flutter create standard

### CODE REVIEW

**Comment by DebkanchanSamadder** [comment]
Couldn't upgrade example to null safety because `location_permissions` on which example app relies on is not null safe

**Comment by DebkanchanSamadder** [comment]
@mehdok

**Comment by mehdok** [comment]
Hi @DebkanchanSamadder 

Thank you for your contribution;

File: example/ios/Runner/AppDelegate.swift [hunk 1]
```diff
@@ -1,33 +1,13 @@
...
-        BackgroundLocatorPlugin.setPluginRegistrantCallback(registerPlugins)
-
-        registerOtherPlugins()
-
-        return super
-            .application(application,
-                         didFinishLaunchingWithOptions: launchOptions)
-    }
-
-    func registerOtherPlugins() {
```
Comment (mehdok) [inline example/ios/Runner/AppDelegate.swift:27]: Why did you remove this?

**Review by mehdok — CHANGES_REQUESTED** [review]

File: example/ios/Runner/AppDelegate.swift (hunk 1 above)
Comment (DebkanchanSamadder) [reply example/ios/Runner/AppDelegate.swift:27]: It was removed due to running `flutter create .` in example when iOS Build was failing. I will add it again

Comment (DebkanchanSamadder) [reply example/ios/Runner/AppDelegate.swift:27]: Checked out the documentation and example for path_provider and to me it seems manually registering the plugin is not necessary however I don't know anything about iOS dev so I can add it if it's needed

Comment (DebkanchanSamadder) [reply example/ios/Runner/AppDelegate.swift:27]: @mehdok

Comment (mehdok) [reply example/ios/Runner/AppDelegate.swift:27]: It is necessary for using other plugins in feedback. Same as Android.

File: example/android/app/src/main/kotlin/rekab/app/example/MainActivity.kt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package rekab.app.example
```
Comment (mehdok) [inline example/android/app/src/main/kotlin/rekab/app/example/MainActivity.kt:1]: I believe this file is useless. please remove it;

**Review by mehdok — CHANGES_REQUESTED** [review]

File: example/ios/Runner/AppDelegate.swift (hunk 1 above)
Comment (DebkanchanSamadder) [reply example/ios/Runner/AppDelegate.swift]: @mehdok Updated in latest commit. Please review further

File: example/android/app/src/main/kotlin/rekab/app/example/MainActivity.kt (hunk 2 above)
Comment (DebkanchanSamadder) [reply example/android/app/src/main/kotlin/rekab/app/example/MainActivity.kt]: removed in latest commit

**Review by DebkanchanSamadder — COMMENTED** [review]
sorry this was a mistake
